src/ContractCallDataEncoder.cpp

```diff
@@ -312,6 +312,8 @@ QJsonArray ContractCallDataEncoder::decodeArray(SolidityType const& _type, bytes
 	if (_type.type == QSolidityType::Type::Bytes || _type.type == QSolidityType::Type::String)
 	{
 		bytesConstRef value(_value.data() + (static_cast<size_t>(offset) + 32), static_cast<size_t>(count));
+		if (value.size() < count)
```

Comment:
There are more like these above. Actually every time you create a `bytesConstRef`.